src/main.rs

```diff
@@ -512,3 +538,246 @@ fn setup() -> Opts {
+        let mut opts_testcase = call_opts(&dir, &seqs, &test_pattern, false, None);
+        // TODO: Automatically set count as t/f depending on opts.output
+        opts_testcase.count = true;
+        let result = fqgrep_from_opts(&mut opts_testcase);
+        assert_eq!(result.unwrap(), 2)
+    }
+
+    /// Tests two files (paired reads) for seqs starting with either A or G
+    ///
+    /// Workflow
+    /// Define a temp dir
+    /// Define the sequences to search in
+    /// Define the pattern to seach for
+    /// Call call_opts() and pass resulting StructOpts instance to fqgrep_from_opts()
+    /// Set opts.testcase.paired to true
+    /// Assert that fqgrep() returns two matches
+    ///
+    #[test]
+    fn test_multiple_fq() {
+        let dir = TempDir::new().unwrap();
+        let seqs = vec![
+            vec!["AAGTCTGAATCCATGGAAAGCTATTG", "GGGTCTGAATCCATGGAAAGCTATTG"],
+            vec!["AAGTCTGAATCCATGGAAAGCTATTG", "GGGTCTGAATCCATGGAAAGCTATTG"],
+        ];
+        let test_pattern = vec![String::from("^A"), String::from("^G")];
+        let mut opts_testcase = call_opts(&dir, &seqs, &test_pattern, true, None);
+        opts_testcase.count = true;
+        opts_testcase.paired = true;
+        let result = fqgrep_from_opts(&mut opts_testcase);
+        assert_eq!(result.unwrap(), 2)
+    }
+
+    /// Tests a two fastqs for a seq that starts with A
+    ///
+    /// Workflow
+    /// Set io as io from fgoxide::io::Io
+    /// Define a temp dir and add an output path to the dir
+    /// Clone the output path so it can be read from
+    ///
+    /// Define the sequences to search in
+    /// Define the pattern to seach for
+    /// Call call_opts() and pass resulting StructOpts instance to fqgrep_from_opts()
+    /// Assert that fqgrep() returns two matches
+    ///
+    #[test]
+    fn test_get_output() {
+        let io = Io::default();
+        let dir: TempDir = TempDir::new().unwrap();
+        let out_path = dir.path().join(String::from("output.fq"));
+        // Clone path to read from below
+        // TODO: refactor
+        let x = &out_path.clone();
+        let seqs = vec![
+            vec!["AAGTCTGAATCCATGGAAAGCTATTG"],
+            vec!["GGGTCTGAATCCATGGAAAGCTATTG"],
+        ];
+        let test_pattern = vec![String::from("^A")];
+        let mut opts_testcase = call_opts(&dir, &seqs, &test_pattern, true, Some(out_path));
+        // run fqgrep
+        let _result = fqgrep_from_opts(&mut opts_testcase);
+
+        // read seqs back from the output path
+        let mut lines = io.read_lines(&x).unwrap();
+        // Get only the seqs (every 4th line skipping the first line)
+        let mut count = -2;
+        lines.retain(|_| {
+            count += 1;
+            return count % 4 == 0;
+        });
+        // make expected
+        let expected = vec!["AAGTCTGAATCCATGGAAAGCTATTG"];
+        // compare
+        assert_eq!(lines, expected);
+    }
```

Comment:
Update to use slurp_output()
```suggestion
        let return_sequences = slurp_output(result_path.to_path_buf());
        let expected_sequences = vec![
            ("AAGTCTGAATCCATGGAAAGCTATTG"),
            ("GGGTCTGAATCCATGGAAAGCTATTG"),
        ];
        assert_eq!(expected_sequences, return_sequences);
    }
```